### ISSUE TITLE
Can BRAKER3 delete .sam file when complete samtools sort step?

### ISSUE THREAD

**Issue by changchuanjun** [Gaius-Augustus/BRAKER#778]
Hello, thanks your team for developing BRAKER pipeline accessions which promote community to finish gene structure annotation. When I ran BRAKER3(v) pipeline to annotate my genome,I found this programme could not delete **.sam** file when complete **samtools sort** step which can convert sam file to bam file. sam file is too large and occupies too much space(dozens of Gb).

**Comment by KatharinaHoff** [comment]
If you deal with huge amounts of RNA-Seq, you can align, convert, merge, and sort before providing the merged BAM file to BRAKER. That's how I usually do it. 

Note that adding more and more RNA-Seq does not necessarily improve results. I sometimes alternatively skip technical and biological replicates...

**Comment by changchuanjun** [comment]
@KatharinaHoff , Thank you for your response. When aligning my RNA-seq data to the corresponding soft-masked genome, I noticed only a slight difference compared to the hard-masked genome. Is it correct?

**Comment by KatharinaHoff** [comment]
Yes the results will differ. They may even differ drastically. The aligners
usually ignore softmasking, it's like unmasked.

**Comment by changchuanjun** [comment]
But in my result,they are  only a slight difference

**Comment by HLiYa** [comment]
> If you deal with huge amounts of RNA-Seq, you can align, convert, merge, and sort before providing the merged BAM file to BRAKER. That's how I usually do it.如果你处理大量 RNA-Seq 数据，可以在将合并的 BAM 文件提供给 BRAKER 之前进行比对、转换、合并和排序。我通常就是这样做的。
> 
> Note that adding more and more RNA-Seq does not necessarily improve results. I sometimes alternatively skip technical and biological replicates...请注意，不断增加 RNA-Seq 数据并不一定能改善结果。我有时会交替跳过技术和生物学重复...

Hello, I'm getting an error when using 5 bam files (SCL-001L,001B,00H,001K,001G) generated after hisat2 and samtools sort processing for braker3 annotation. The samtools.sort.SCL-001K.stderr shows:
[bam_sort_core] merging from 1 files and 29 in-memory blocks...
samtools sort: failed writing to "/root/autodl-tmp/braker_output/result/GeneMark-ETP/etp_data/SCL-001K.bam": No such file or directory
But when I check workingdir/GeneMark-ETP/etp_data, there are only three files: SCL-001L.bam, 001B.bam, and 00H.bam. Why is this?
Then I merged these 5 bam files using samtools merge to generate merged.bam, and used merged.bam as input for braker3, but still encountered:
(/root/autodl-tmp/braker_tools) root@autodl# cat /root/autodl-tmp/braker_output/result/errors/samtools.sort.merged.stderr
samtools sort: failed writing to "/root/autodl-tmp/braker_output/result/GeneMark-ETP/etp_data/merged.bam.tmp.0004.bam": No such file or directory

**Comment by KatharinaHoff** [comment]
[BRAKER4](https://github.com/Gaius-Augustus/BRAKER4) natively supports short-read RNA-Seq (BAM or FASTQ), PacBio IsoSeq (BAM or FASTQ, aligned with minimap2), and a dual mode combining both with proteins. All inputs are specified via a samples.csv file. BAM sorting, temporary file cleanup, and multi-library handling are managed by Snakemake rules.
